src/query/execute_statement.c

```diff
@@ -17775,13 +17775,21 @@ do_alter_synonym (PARSER_CONTEXT * parser, PT_NODE * statement)
   else
     {
       /* PT_SYNONYM_TARGET_NAME (statement) != NULL */
+      if (PT_SYNONYM_DBLINKED (statement))
+	{
+	  sprintf (target_name_buf, "%s", (PT_NAME_ORIGINAL (PT_SYNONYM_TARGET_NAME (statement))));
+	  /* target_owner = synonym_owner */
+	  target_owner_obj = db_find_user (PT_NAME_ORIGINAL (PT_SYNONYM_OWNER_NAME (statement)));
```

Comment:
In the pt_set_user_specified_name function, target_owner_name is set to synonym_owner_name.
So the branch to set target_owner_obj is unnecessary.
```
if (PT_SYNONYM_DBLINKED (node))
  {
    PT_SYNONYM_TARGET_OWNER_NAME (node) = pt_name (parser, synonym_owner_name);
  }
```

Reply:
the target_obj is necessary for checking the owner in case of "not dblinked"